embassy-nrf/src/pdm.rs

```diff
@@ -362,6 +378,125 @@ impl<'d, T: Instance> Pdm<'d, T> {
         drop.defuse();
         Ok(())
     }
+
+    /// Continuous sampling with dynamic double buffers.
+    ///
+    /// A sampler closure is provided that receives the buffer of samples.
+    /// The closure returns a command indicating if the sampling
+    /// should continue or stop.
+    ///
+    /// This version of the function is for allowing zero-copy, dynamic double buffering.
+    /// The bufers are accepted as a tuple of raw pointers, lengths and capacities matching
+    /// the Vec::from_raw_parts() function.
+    ///
+    /// SAFTEY:
+    /// It is expected that the caller will set the length to non-zero and re-use the same
+    /// length and capacity for each buffer. Changing buffer sizes is not supported.
+    ///
+    /// At least one buffer will still be allocated when the function returns. The caller
+    /// should call free it to prevent leaks.
```

Comment:
Maybe it makes sense to add some assertions to ensure this invariant is met? (Non-zero at least).

ALso SAFTEY -> SAFETY